### PR TITLE
feat(js-config): add node 20 lambda configurations

### PR DESCRIPTION
**Pull request by thib3113**
## Description

 - Add node 20 lambda tsconfig
 - add strict flag by default
 - move devDependencies to dependencies (so, dependant projects doesn't need to install all of them separately) + this package is already a devDependencies (so not included in bundle)

<h2> How this pull request can be tested ? </h2>

on a lambda 

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Comment by thib3113** [comment]
@LoThomo / @Med1233 a review please ?

**Comment by LoThomo** [comment]
Have you tested building the Centreon web application with the dev ecosystem with this changes?

**Comment by thib3113** [comment]
> Have you tested building the Centreon web application with the dev ecosystem with this changes?

Yes, I just test it on dev-ecosystem . 

I've updated the js-config configuration in : 

```
centreon-dev-ecosystem
centreon-anomaly-detection/www/modules/centreon-anomaly-detection/react
centreon-autodiscovery/tests/e2e
centreon-autodiscovery/www/modules/centreon-autodiscovery-server/react
centreon-bam/tests/e2e
centreon-bam/www/modules/centreon-bam-server/react
centreon-cloud-extensions/www/modules/centreon-cloud-extensions/react
centreon-it-edition-extensions/www/modules/centreon-it-edition-extensions/react
centreon-license-manager/www/modules/centreon-license-manager/react
centreon-map/web/react
``` 

And started them . it can start successfully,

**Review by LoThomo — DISMISSED** [review]

File: centreon/packages/js-config/package.json [hunk 1]
```diff
@@ -1,60 +1,63 @@
...
-    "cypress-multi-reporters": "^1.6.4",
-    "cypress-terminal-report": "^5.3.9",
-    "dockerode": "^4.0.0",
-    "dotenv": "^16.3.1",
-    "esbuild": "^0.19.5",
-    "mochawesome": "^7.1.3"
-  }
+    "name": "@centreon/js-config",
+    "description": "Centreon Frontend shared build configuration",
+    "version": "24.4.0",
```
Comment (thib3113) [inline centreon/packages/js-config/package.json:4]:
```suggestion
    "version": "24.4.1-lambda-node20-jsconfig.6",
```

File: centreon/packages/js-config/package.json [hunk 2]
```diff
@@ -1,60 +1,63 @@
...
-    "cypress-multi-reporters": "^1.6.4",
-    "cypress-terminal-report": "^5.3.9",
-    "dockerode": "^4.0.0",
-    "dotenv": "^16.3.1",
-    "esbuild": "^0.19.5",
-    "mochawesome": "^7.1.3"
-  }
+    "name": "@centreon/js-config",
+    "description": "Centreon Frontend shared build configuration",
+    "version": "24.4.1",
```
Comment (Thebarda) [inline centreon/packages/js-config/package.json:4]:
```suggestion
    "version": "24.4.0",
```

**Review by LoThomo — APPROVED** [review]

**Review by sc979 — APPROVED** [review]
LGTM on pipeline part

**Pull request merged**